src/device-api/v1.ts

```diff
@@ -106,52 +106,54 @@ export function createV1Api(router: express.Router) {
 	router.post('/v1/apps/:appId/stop', createV1StopOrStartHandler('stop'));
 	router.post('/v1/apps/:appId/start', createV1StopOrStartHandler('start'));
 
-	router.get('/v1/apps/:appId', (req: AuthorizedRequest, res, next) => {
+	router.get('/v1/apps/:appId', async (req: AuthorizedRequest, res, next) => {
 		const appId = checkInt(req.params.appId);
 		eventTracker.track('GET app (v1)', { appId });
 		if (appId == null) {
 			return res.status(400).send('Missing app id');
 		}
-		return Promise.join(
-			applicationManager.getCurrentApps(),
-			applicationManager.getStatus(),
-			function (apps, status) {
-				const app = apps[appId];
-				const service = app?.services?.[0];
-				if (service == null) {
-					return res.status(400).send('App not found');
-				}
 
-				if (app.services.length > 1) {
-					return res
-						.status(400)
-						.send(
-							'Some v1 endpoints are only allowed on single-container apps',
-						);
-				}
-
-				// handle the case where the appId is out of scope
-				if (!req.auth.isScoped({ apps: [app.appId] })) {
-					res.status(401).json({
-						status: 'failed',
-						message: 'Application is not available',
-					});
-					return;
-				}
-
-				// Don't return data that will be of no use to the user
-				const appToSend = {
-					appId,
-					commit: status.commit!,
-					containerId: service.containerId,
-					env: _.omit(service.config.environment, constants.privateAppEnvVars),
-					imageId: service.config.image,
-					releaseId: service.releaseId,
-				};
-
-				return res.json(appToSend);
-			},
-		).catch(next);
+		try {
+			const apps = await applicationManager.getCurrentApps();
+			const app = apps[appId];
+			const service = app?.services?.[0];
+			if (service == null) {
+				return res.status(400).send('App not found');
+			}
+
+			if (app.services.length > 1) {
+				return res
+					.status(400)
+					.send('Some v1 endpoints are only allowed on single-container apps');
+			}
+
+			// handle the case where the appId is out of scope
+			if (!req.auth.isScoped({ apps: [app.appId] })) {
+				res.status(401).json({
+					status: 'failed',
+					message: 'Application is not available',
+				});
+				return;
+			}
```

Comment:
This check should happen BEFORE the service count check. It's not a huge issue since a _valid_ key is required to get this far, but technically we're leaking information to authenticated, but unauthorised, clients.